azure-pipelines.yml

```diff
@@ -1,54 +1,168 @@
-
-- task: PublishBuildArtifacts@1
-  displayName: Publish Artifacts  
-  inputs:
-    pathToPublish: $(Build.ArtifactStagingDirectory)\Packages
-    artifactType: container
-    artifactName: Package
+- master
+- rel/*
+
+jobs:
+- job: Windows
+  pool:
+    vmImage: windows-2019
+
+  variables:
+    BuildConfiguration: Release
+
+  steps:
+  - task: DotNetCoreCLI@2  
+    inputs:
+      command: custom
+      custom: tool
+      arguments: install --tool-path . nbgv
+    displayName: Install NBGV tool
+
+  - script: nbgv cloud
+    displayName: Set Version
+
+  - powershell: |
+      mkdir $Env:Temp\Packages -Force
+    displayName: Create packages temp folder
+
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: pack
+      packagesToPack: Source/MSBuild.Sdk.Extras/MSBuild.Sdk.Extras.csproj
+      configuration: $(BuildConfiguration)
+      packDirectory: $(Build.ArtifactStagingDirectory)\Packages    
+      verbosityPack: Minimal
+    displayName: Build Package
+
+  - task: PowerShell@2
+    displayName: Authenticode Sign artifacts
+    inputs:
+      filePath: Tools\Sign-Package.ps1
+    env:
+      SignClientUser: $(SignClientUser)
+      SignClientSecret: $(SignClientSecret)
+      ArtifactDirectory: $(Build.ArtifactStagingDirectory)\Packages
+    condition: and(succeeded(), not(eq(variables['build.reason'], 'PullRequest')), not(eq(variables['SignClientSecret'], '')), not(eq(variables['SignClientUser'], '')))
+
+  - task: PublishBuildArtifacts@1
+    displayName: Publish Artifacts  
+    inputs:
+      pathToPublish: $(Build.ArtifactStagingDirectory)\Packages
+      artifactType: container
+      artifactName: Package
+
+- job: Windows_Tests
+  pool:
+    vmImage: windows-2019
+
+  variables:
+    BuildConfiguration: Release
+    TMPDIR: $(Build.ArtifactStagingDirectory)
+    DisableNerdBank: true
+    PackageVersion: 42.42.42
+
+  steps:
+  - powershell: |
+      mkdir $(Build.ArtifactStagingDirectory)\Packages -Force
+    displayName: Create packages temp folder
+
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: pack
+      packagesToPack: Source/MSBuild.Sdk.Extras/MSBuild.Sdk.Extras.csproj
+      configuration: $(BuildConfiguration)
+      packDirectory: $(Build.ArtifactStagingDirectory)\Packages    
+      verbosityPack: Minimal
+    displayName: Build Package
+
+  - task: MSBuild@1
+    inputs:
+      solution: '$(build.sourcesdirectory)\TestProjects\Windows-Desktop-C#\Windows-Desktop.sln'
+      msbuildArguments: /r /p:SignManifests=false
+      configution: $(BuildConfiguration)
+    displayName: MSBuild Test Projects 1
+
+  - task: MSBuild@1
+    inputs:
+      solution: '$(build.sourcesdirectory)\TestProjects\Linux-C#\netstd2Library\netstd2Library.csproj'
+      msbuildArguments: /r
+      configution: $(BuildConfiguration)
+    displayName: MSBuild Test Projects 2
+
+- job: Linux
+  container: nventive/wasm-build:1.4.1
```

Comment:
Should this be pinned to a specific version or is there a 'latest' tag that should be used?

Reply:
This is pinned, as I wanted to make sure it tests that version of msbuild (16.0), later builds will get newer versions. The point is to understand how far compatibility goes, and if it breaks a specific version, at least it is know and documented.